File: benchmarks/successor/representation-v1/transformed/descriptive/csharp/gold/008-summary-api/Program.cs
using System.Text.Json;

namespace OrderFlow;

public static class Program
{
    private static readonly JsonSerializerOptions Options = new()
    {
        PropertyNameCaseInsensitive = true,
        PropertyNamingPolicy = JsonNamingPolicy.CamelCase
    };

    public static void Main()
    {
        string? line;
        while ((line = Console.ReadLine()) is not null)
        {
            try
            {
                var request = JsonSerializer.Deserialize<Request>(line, Options)
                    ?? throw new InvalidOperationException("Request was null");
                var response = OrderFlowEngine.Handle(request);
                Console.WriteLine(JsonSerializer.Serialize(response, Options));
            }
            catch (Exception ex)
            {
                Console.WriteLine(JsonSerializer.Serialize(new { error = ex.Message }, Options));
            }
        }
    }
}

File: benchmarks/successor/representation-v1/transformed/descriptive/fsharp/gold/008-summary-api/Program.fs
open System
open System.Text.Json

let options =
    JsonSerializerOptions(
        PropertyNameCaseInsensitive = true,
        PropertyNamingPolicy = JsonNamingPolicy.CamelCase
    )

let mutable running = true

while running do
    match Console.ReadLine() with
    | null -> running <- false
    | input ->
        try
            let request =
                JsonSerializer.Deserialize<OrderFlowEngine.Request>(input, options)

            match request with
            | null -> raise (InvalidOperationException("Request was null"))
            | request ->
                let response = OrderFlowEngine.handle request
                Console.WriteLine(JsonSerializer.Serialize(response, options))
        with ex ->
            Console.WriteLine(JsonSerializer.Serialize({| error = ex.Message |}, options))
